base/intfuncs.jl

```diff
@@ -325,7 +325,8 @@ end
 
 function digits!{T<:Integer}(a::AbstractArray{T,1}, n::Integer, base::Integer=10)
     2 <= base || throw(ArgumentError("base must be ≥ 2, got $base"))
-    base - 1 <= typemax(T) || throw(ArgumentError("type $T too small for base $base"))
+    !applicable(typemax, T) ||
```

Comment:
OK so let's not hold our breath. I just tried a new version where I replace `applicable` by `hastypemax` defined as follows:
```
hastypemax{T<:Base.BitInteger}(::Type{T}) = true
hastypemax{T<:BigInt}(::Type{T}) = false
hastypemax(::Type{T}) = applicable(typemax, T)
```
The perfs seem to be similar to the base version, maybe few percents behind (`hastypemax` may be not removed at compilation time?). But it's a  bit ugly, I'd rather wait for `applicable` (or `method_exists`) to  be improved, and pay 30% penalty in the meantime. But if someone thinks it's important to preserve every bit of efficiency in `digits!`, I will update with this `hastypemax` function.